dynamic/Dockerfile

```diff
@@ -1,29 +1,37 @@
-FROM ubuntu:21.04
+# syntax = docker/dockerfile:1.2
 
-ARG DEBIAN_FRONTEND=noninteractive
-ARG TZ=Etc/UTC
+FROM registry.access.redhat.com/ubi8/ubi:latest
 
-# Setup of dependencies we need inside the container
-RUN apt-get update && \
-    apt-get install -y --no-install-recommends gnupg2 ca-certificates curl && \
-    echo "deb [arch=amd64] https://download.docker.com/linux/ubuntu focal stable" > /etc/apt/sources.list.d/docker.list && \
-    apt-key adv --fetch-keys "https://download.docker.com/linux/ubuntu/gpg" && \
-    apt-get update && \
-    apt-get install -y --no-install-recommends inotify-tools docker-ce-cli containerd python3-pip iptables iproute2 jq && \
-    apt-get upgrade -y && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/*
+LABEL name="instana-agent" \
+      vendor="Instana Inc." \
+      release="1" \
+      summary="Instana APM agent" \
+      description="Instana APM agent"
 
-# Setup to install the Instana package
-RUN echo "deb [arch=amd64] https://_:${FTP_PROXY}@packages.instana.io/agent/deb generic main" > /etc/apt/sources.list.d/instana-agent.list && \
-    apt-key adv --fetch-keys "https://packages.instana.io/Instana.gpg" && \
-    apt-get update && \
-    apt-get install -y --no-install-recommends gomplate && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/*
+ADD docker.repo /etc/yum.repos.d/docker.repo
+
+RUN --mount=type=secret,id=download_key,dst=/run/secrets/download_key,required=true \
```

Comment:
The builds will need to be updated for this to work as well. Is there no backwards-compatible way?
(at the least there must be a TODO so we know after merging this to update Jenkins builds)

Reply:
Build needs fixing, yes, but IMO it is worth it :-)